test/module/shared_model/interface_test.cpp

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+#include "logger/logger.hpp"
+
+#include "builders/protobuf/transaction.hpp"
+
+class TransactionFixture : public ::testing::Test {
+ public:
+  TransactionFixture()
+      : keypair(shared_model::crypto::DefaultCryptoAlgorithmType::
+                    generateKeypair()),
+        time(iroha::time::now()) {}
+
+  shared_model::crypto::Keypair keypair;
+  shared_model::interface::types::TimestampType time;
+
+  logger::Logger log = logger::log("TransactionFixture");
+
+  auto makeTx() {
+    log->info("keypair = {}, timestemp = {}", keypair.toString(), time);
+    return std::make_shared<shared_model::proto::Transaction>(
+        shared_model::proto::TransactionBuilder()
```

Comment:
Is it possible to rewrite the test with mock transaction?

Reply:
Unfortunately, not. Because unordered_set of Signatures holds in implementation.

Reply:
Also, additional research is showed that with the mock approach we have to write much boilerplate code. In particular, almost-effective implementation of `addSiganture` and addition `hash(MockSignature)` and `operator<` for mockSignature type.